format.go

```diff
@@ -88,6 +100,25 @@ func (v *Value) Format() {
 	v.UpdateOffsets()
 }
 
+// Range iterates through a Value breadth-first and applies
+// a given method on each leaf.
+func (v *Value) Range(f func(v *Value) bool) {
+	switch v2 := v.Value.(type) {
+	case Literal:
+		f(v)
+	case composite:
+		v2.rangeValues(f)
+		for i := 0; i < v2.length(); i++ {
+			v2.valueAt(i).Range(f)
+		}
+	}
+}
+
+// SetValue mutates the value.
+func (v *Value) SetValue(vt ValueTrimmed) {
+	v.Value = vt
+}
```

Comment:
this is assigning only an exported field. Any reason for the method? Is it satisfying some interface, or?

Reply:
Javascript habits 😅 removed